### PR TITLE
Post Reactions

### LINKED ISSUE #542: Improving and Adding some card functionality on the feeds page
## Problem to be solved

There is a need to add some functionality to the cards and improve its UI in terms of sizing.
We need to add the functionality of "**Like**", "**Happy**", "**Favourite**", "**Donut**(meaning bookmark)" and clicking on respective icons it should do the following things:
- Like: Should like that particular post written by the author
- Happy: This will let the user know to give a reaction of being glad to read that
- Favourite: Clicking on "Heart" button should add it within the favourites option
- Donut: Will book mark up that particular post

Also, there should be three dots added upon the card right side which on clicking will show the options to "Edit, Share, and Delete"

For more reference please check the image below:
Group 280

Design of this card can be found here - https://www.figma.com/file/iZvb7rLYRp3Di3wILVQ7V9/CodeUino-Donut?node-id=2780%3A104

### PR DESCRIPTION
**Pull request by AuraOfDivinity**
Fixes #542 

- [x] Implements animated Reactions below posts as requested.
- [x] Implements a dropdown on the top right corner of the Post card.

Reactions

This is only the frontend implementation. Some changes to the backend might be required for other reactions (Heart, Happy and Donut) since only upvotes are supported at the moment

### CODE REVIEW

**Comment by AuraOfDivinity** [comment]
@jaskiratsingh2000 A few questions regarding this on my end,

1. I think we should separate the user reactions to a particular post from other functionalities such as "Adding to favorite" and bookmarking a post. We could have these functionalities on the top right dropdown for instance. 

2. How will we displaying the reactions to a particular post? Maybe we could have a separate modal to display just like on Facebook? Something like the following image? 

image

**Comment by jaskiratsingh2000** [comment]
Thanks for taking this up. Well, I think we could do in a same way as it is
showed inside the facebook

File: package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "chart.js": "^2.9.3",
     "dotenv": "^8.2.0",
     "env-cmd": "^10.1.0",
+    "framer-motion": "^2.3.0",
```
Comment (Rupeshiya) [inline package.json:20]: @AuraOfDivinity I am not happy with this npm package, as it has 8 dependencies, can't do it with css transitions??

File: src/user/dashboard/news-feed/popups/PostReactionsModal.js [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+      <div className="modal__buttons">
+        <Button onClick={props.onHide} className="modal__save">
+          <span className="modal__buttontext">Exit</span>
+        </Button>
+      </div>
+    </Modal>
+  );
+};
+PostReactionModal.propTypes = {
+  onClick: PropTypes.func,
```
Comment (Rupeshiya) [inline src/user/dashboard/news-feed/popups/PostReactionsModal.js:148]: onClick is not passed.

Comment (Rupeshiya) [reply src/user/dashboard/news-feed/popups/PostReactionsModal.js:148]: Also connect to state, you forgot to connect

**Comment by Rupeshiya** [comment]
Also @AuraOfDivinity Please add **Pin icon,** already told earlier as well.

File: src/actions/postAction.js [hunk 3]
```diff
@@ -1,95 +1,101 @@
...
 
 // GET ALL POSTS
 export const getAllPosts = (pagination = 10, page = 1) => async (dispatch) => {
   try {
-    const res = await axios.get(`${BASE_URL}/post/all_posts?pagination=${pagination}&page=${page}`)
-    dispatch(setRequestStatus(false))
-    if(res.status === 200) {
-      dispatch(setRequestStatus(true))
-      console.log('all posts ', res.data.posts)
+    const res = await axios.get(
```
Comment (Rupeshiya) [inline src/actions/postAction.js:10]: @AuraOfDivinity Please remove these unnecessary format changes, this is pretty annoying issue

File: src/actions/postAction.js [hunk 4]
```diff
@@ -1,95 +1,101 @@
...
+export const getAllPinnedPosts = (pagination = 10, page = 1) => async (
+  dispatch
+) => {
   try {
-    const res = await axios.get(`${BASE_URL}/post/all/pinned?pagination=${pagination}&page=${page}`)
-    dispatch(setRequestStatus(false))
-    if(res.status === 200){
-      dispatch(setRequestStatus(true))
-      console.log('fetching all pinned posts ', res.data.pinnedPost)
+    const res = await axios.get(
```
Comment (Rupeshiya) [inline src/actions/postAction.js:32]: un-necessary changes

**Comment by AuraOfDivinity** [comment]
Updated reactions.

- [x] Added pin post icon
- [x] Added PostReactions Modal
- [x] Added text based on the number of reactions at the bottom of the component.

ReactionsUpdated

**Pull request merged**